src/main/java/net/openhft/chronicle/queue/impl/single/SCQIndexing.java

```diff
@@ -726,7 +727,12 @@ public long lastSequenceNumber(@NotNull ExcerptContext ec, boolean approximate)
                     continue;
                 if (sequence == Sequence.NOT_FOUND)
                     break;
-                return sequence;
+                try {
+                    Wire wireForIndex = ec.wireForIndex();
+                    return wireForIndex == null ? sequence : linearScanByPosition(wireForIndex, Long.MAX_VALUE, sequence, address, true);
```

Comment:
This is the change, when we get a valid writePosition/sequence pair, we do a linear scan to make sure it's accurate. This covers off the case where we suffered a partial update.

It also removes the need to have separate approximate/exact implementations of `lastSequenceNumber`.

The ONE implementation just does the following
- read the writePosition/sequence header
- if it's consistent, do a quick linear scan to ensure it's up-to-date (this linear scan should be very short, at most 2 excerpts
- if it's not consistent, traverse the index and roll cycle contents to determine the last sequence